src/tests/test_config.cpp

```diff
@@ -286,4 +289,51 @@ BOOST_AUTO_TEST_CASE ( test_variable_info )
 	}
 }
 
+BOOST_AUTO_TEST_CASE (
+		add_child_ThisIsEmpty_newKey_AppendEmptyChildToThis_ReturnNewChild )
+{
+	config actual;
+	const config new_child = actual.add_child("A");
+	const config expected("A");
+	// Assert add_child adds the child correctly
+	BOOST_CHECK_EQUAL (actual, expected);
+	// Assert the new child returned by config::add_child is correct
+	BOOST_CHECK_EQUAL (new_child, config());
+}
+
+namespace bdata = boost::unit_test::data;
+BOOST_DATA_TEST_CASE (
+		add_child_ThisIsNotEmpty_newOrExistingKey_lOrRValue_AppendNewChildToThis_ReturnNewChild,
+		bdata::make<std::string>({"A", "B", "C"})
+		* bdata::make<std::string>({"lvalue_ref", "rvalue_ref"}),
+		key, update_ref_kind)  // 3 * 2 = 6 cases
+{
+	// Data for testing base.add_child(key, update)
+	const config base {"A", config(), "a", 1};   // [A][/A] a = 1
+	const config update {"B", config(), "b", 2}; // [B][/B] b = 2
+	// Make expected
```

Comment:
Please move the after-the-semicolon comments to this line, so it's more like

    // Expected structure is [A][/A] a=1 [new_key][B][/B] b=2 [/new_key]